modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpNextgenClientCodegen.java

```diff
@@ -154,4 +162,41 @@ private ModelsMap postProcessModelsMap(ModelsMap objs) {
         }
         return objs;
     }
+
+    @Override
+    public OperationsMap postProcessOperationsWithModels(OperationsMap objs, List<ModelMap> allModels) {
+        objs = super.postProcessOperationsWithModels(objs, allModels);
+        OperationMap operations = objs.getOperations();
+        for (CodegenOperation operation : operations.getOperation()) {
+            if (operation.returnType == null) {
+                operation.vendorExtensions.putIfAbsent("x-php-return-type", "void");
+            }
+            else {
+                operation.vendorExtensions.putIfAbsent("x-php-return-type", operation.returnType);
+            }
+
+            for (CodegenParameter param : operation.allParams) {
+                if (param.isArray || param.isMap) {
+                    param.vendorExtensions.putIfAbsent("x-php-param-type", "array");
+                }
+                else {
+                    param.vendorExtensions.putIfAbsent("x-php-param-type", param.dataType);
+                }
+            }
+        }
+
+        return objs;
+    }
+
+    @Override
+    public String toDefaultValue(CodegenProperty codegenProperty, Schema schema) {
+        if (codegenProperty.isArray) {
+            if (schema.getDefault() != null) {
+                return "[" + schema.getDefault().toString() + "]";
```

Comment:
@JulianVennen @wing328 I wonder whether we need to consider an edge case here: What if the default value isn't scalar but an empty array? From my understanding, `default: []` complies with the OpenAPI spec and is a valid use case.

Consider the following components:
```json
{
  "properties": {
    "myArray": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "default": []
    }
  },
  "type": "object"
}
```

In its current form, the generated constructor method resorts to a nested array in case `my_array` is not  specified:

```php
    public function __construct(array $data = null)
    {
        $this->setIfExists('my_array', $data ?? [], [[]]);
    }
```

This isn't the result I would have expected TBH. Could you shed some light on this, please?